s3/s3_operations.cpp

```diff
@@ -998,6 +998,14 @@ namespace irods_s3 {
             //  do a stat with a retry if not found
             if (s3_transport_ptr->is_last_file_to_close() && result.ok()) {
 
+                // reset global variables
```

Comment:
Consider expanding on this.

It seems important for future readers to know why these need to be reset.

Reply:
done